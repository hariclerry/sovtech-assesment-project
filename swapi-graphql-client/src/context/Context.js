import React, { createContext, useState } from "react";
export const SearchContext = createContext();

export const SearchProvider = (props) => {
  const [isSearching, setIsSearching] = useState(false);

  return (
    <SearchContext.Provider value={{ isSearching, setIsSearching }}>
      {props.children}
    </SearchContext.Provider>
  );
};
