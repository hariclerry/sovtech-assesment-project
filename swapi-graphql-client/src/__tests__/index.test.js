import { render, cleanup, waitFor } from '@testing-library/react'
import { MockedProvider } from '@apollo/client/testing'
import { SearchProvider } from "../context/Context"
import PeopleLists, { GET_PAGE_PEOPLE } from '../components'

const mocks = [
    {
        request: {
            query: GET_PAGE_PEOPLE,
            variables: {
                page: 1,
            },
            result: {
                data: {
                    count: 9,
                    next: "",
                    previous: "",
                    results: [
                        {
                            name: "Luke Skywalker",
                            height: 172,
                            mass: "77",
                            gender: "male",
                        },
                        {
                            name: "C-3PO",
                            height: 167,
                            mass: 75,
                            gender: "n/a",
                        },
                    ]
                },
            },
        }
    }
]

describe('People', () => {
    afterEach(cleanup)
    test('renders all todo elements', async () => {
        const { container } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <SearchProvider>
                    <PeopleLists />
                </SearchProvider>
            </MockedProvider>,
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        expect(container).toMatchSnapshot();
    });
})