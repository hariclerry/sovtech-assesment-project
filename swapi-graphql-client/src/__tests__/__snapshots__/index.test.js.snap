// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`People renders all todo elements 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-15216nb-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-nc4yk0"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
      >
        <div
          class="css-1orzy9s"
        >
          <div
            class="MuiInputBase-root MuiInputBase-colorPrimary css-vd3me9-MuiInputBase-root"
          >
            <input
              aria-label="search"
              class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
              name="search"
              placeholder="Search…"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-fullWidth MuiButtonBase-root  css-stq1bk-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Submit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <div
      class="makeStyles-root-1 css-1p5q5e5-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 makeStyles-paginate-2 css-ahj2mt-MuiTypography-root"
      >
        Page: 
        1
      </p>
      <nav
        aria-label="pagination navigation"
        class="MuiPagination-root MuiPagination-outlined makeStyles-paginate-2 css-1oj2twp-MuiPagination-root"
      >
        <ul
          class="MuiPagination-ul css-wjh20t-MuiPagination-ul"
        >
          <li>
            <button
              aria-label="Go to previous page"
              class="MuiButtonBase-root Mui-disabled MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-disabled MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateBeforeIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
                />
              </svg>
            </button>
          </li>
          <li>
            <button
              aria-current="true"
              aria-label="page 1"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular Mui-selected MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              1
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 2"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              2
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 3"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              3
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 4"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              4
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to page 5"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              5
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <div
              class="MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-ellipsis css-1v2lvtn-MuiPaginationItem-root"
            >
              …
            </div>
          </li>
          <li>
            <button
              aria-label="Go to page 9"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-page css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              9
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
          <li>
            <button
              aria-label="Go to next page"
              class="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-sizeMedium MuiPaginationItem-outlined MuiPaginationItem-circular MuiPaginationItem-previousNext css-lqq3n7-MuiButtonBase-root-MuiPaginationItem-root"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiPaginationItem-icon css-g2z002-MuiSvgIcon-root-MuiPaginationItem-icon"
                data-testid="NavigateNextIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
